web/src/engine/osk/src/keyboard-layout/oskKey.ts

```diff
@@ -12,6 +12,39 @@ import buttonClassNames from '../buttonClassNames.js';
 import { KeyElement } from '../keyElement.js';
 import VisualKeyboard from '../visualKeyboard.js';
 
+/**
+ * Replace default key names by special font codes for modifier keys
+ *
+ *  @param  {string}  oldText
+ *  @return {string}
+ **/
+export function renameSpecialKey(oldText: string, vkbd: VisualKeyboard): string {
+ // If a 'special key' mapping exists for the text, replace it with its corresponding special OSK character.
+ switch(oldText) {
+   case '*ZWNJ*':
+     // Default ZWNJ symbol comes from iOS.  We'd rather match the system defaults where
+     // possible / available though, and there's a different standard symbol on Android.
+     oldText = vkbd.device.OS == DeviceSpec.OperatingSystem.Android ?
+       '*ZWNJAndroid*' :
+       '*ZWNJiOS*';
+     break;
+   case '*Enter*':
+     oldText = vkbd.isRTL ? '*RTLEnter*' : '*LTREnter*';
+     break;
+   case '*BkSp*':
+     oldText = vkbd.isRTL ? '*RTLBkSp*' : '*LTRBkSp*';
+     break;
+   default:
+     // do nothing.
```

Comment:
Do we also need to handle the new characters added in #9547?

Reply:
I only moved the method so that it's exportable; I made no further changes.  If it worked with the new chars before, it'll work now.